src/providers/cloudstack/network.rs

```diff
@@ -93,3 +97,48 @@ impl MetadataProvider for CloudstackNetwork {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
The other providers put their mock tests in a separate `mock_tests.rs` file.  We should probably do the same.

Reply:
done